packages/ui/src/utilities/buildFormState.ts

```diff
@@ -207,5 +230,82 @@ export const buildFormState = async ({ req }: { req: PayloadRequest }): Promise<
     }
   }
 
-  return result
+  if (returnLockStatus && req.user && (id || globalSlug)) {
```

Comment:
Should there also be another condition in the if condition to prevent this block when locking isn't enabled for the collection/global?